src/services/picture.ts

```diff
@@ -34,4 +38,55 @@ export class PictureService {
       upload.start();
     });
   }
+
+  convertURLtoBlob(URL: string): Promise<Blob> {
+    return new Promise((resolve, reject) => {
+      const image = document.createElement('img');
+
+      image.onload = () => {
+        try {
+          const dataURI = this.convertImageToDataURI(image);
+          const blob = this.convertDataURIToBlob(dataURI);
+
+          resolve(blob);
+        }
+        catch (e) {
+          reject(e);
+        }
+      };
+
+      image.src = URL;
+    });
+  }
+
+  convertImageToDataURI(image: HTMLImageElement): string {
+    // Create an empty canvas element
+    const canvas = document.createElement('canvas');
```

Comment:
The `datauri` package is NodeJS compatible only, and the `image-to-data-uri` is only partially implements the feature since it only converts the image to data URI, while I want to convert it to Blob. In addition, It has a very poor download rate, and I'm not sure how trusted this is. Overall it's a small code snippet that I'd rather not taking any risks with, considering the currently available resources.